src/Dependencies/CodeAnalysis.Metadata/CustomDebugInfoEncoder.cs

```diff
@@ -143,20 +143,59 @@ public void AddDynamicLocals(IReadOnlyCollection<(string LocalName, byte[] Flags
             AddRecord(
                 CustomDebugInfoKind.DynamicLocals,
                 dynamicLocals,
-                (info, builder) =>
+                (infos, builder) =>
                 {
-                    builder.WriteInt32(dynamicLocals.Count);
+                    builder.WriteInt32(infos.Count);
 
-                    foreach (var (name, flags, count, slotIndex) in dynamicLocals)
+                    foreach (var info in infos)
                     {
-                        Debug.Assert(flags.Length == DynamicAttributeSize);
-                        Debug.Assert(name.Length <= IdentifierSize);
-
-                        builder.WriteBytes(flags);
-                        builder.WriteInt32(count);
-                        builder.WriteInt32(slotIndex);
-                        builder.WriteUTF16(name);
-                        builder.WriteBytes(0, sizeof(char) * (IdentifierSize - name.Length));
+                        Debug.Assert(info.Flags.Length <= DynamicAttributeSize);
+                        Debug.Assert(info.LocalName.Length <= IdentifierSize);
+
+                        builder.WriteBytes(info.Flags);
+                        builder.WriteBytes(0, sizeof(byte) * (DynamicAttributeSize - info.Flags.Length));
+                        builder.WriteInt32(info.Count);
+                        builder.WriteInt32(info.SlotIndex);
+                        builder.WriteUTF16(info.LocalName);
+                        builder.WriteBytes(0, sizeof(char) * (IdentifierSize - info.LocalName.Length));
+                    }
+                });
+        }
+
+        public void AddTupleElementNames(IReadOnlyCollection<(string LocalName, int SlotIndex, int ScopeStart, int ScopeEnd, ImmutableArray<string> Names)> tupleLocals)
+        {
+            Debug.Assert(tupleLocals != null);
+
+            AddRecord(
+                CustomDebugInfoKind.TupleElementNames,
+                tupleLocals,
+                (infos, builder) =>
+                {
+                    Debug.Assert(infos.Count > 0);
+
+                    builder.WriteInt32(infos.Count);
+                    foreach (var info in infos)
+                    {
+                        // Constants have slot index -1 and scope specified,
+                        // variables have a slot index specified and no scope.
+                        Debug.Assert((info.SlotIndex == -1) ^ (info.ScopeStart == 0 && info.ScopeEnd == 0));
+
+                        builder.WriteInt32(info.Names.Length);
+                        foreach (var name in info.Names)
+                        {
+                            if (name != null)
+                            {
+                                builder.WriteUTF8(name);
```

Comment:
It's the same. Empty string in UTF8 is encoded as empty sequence of bytes. Which is the same as now writing the name out.